### ISSUE TITLE
TypeError: Cannot set property listening of #<Server> which has only a getter 

### ISSUE THREAD

**Issue by opusvega** [mpneuried/nodecache#110]
I am using Socket.IO and inside the io.on("connection") when I try to set the key's and Values I get the above error but if I set it out side the io.on() it works.

**Comment by erdii** [comment]
Please provide code examples, as I don't really get what you mean.

**Comment by erdii** [comment]
`cache1` and `cache2` are functions and cannot work in both or your examples.

`module.exports.cache1= () =>(cache1) ` is a function that, on execution, returns a reference to `cache1`

So `cache1().set("Key1", "Value1")` should work.

**Comment by opusvega** [comment]
Sorry. The code is actually written the way you mentioned. But the main issue is, It does not work in io.on() and works out side of it. I am getting the error **TypeError: Cannot set property listening of #<Server> which has only a getter at _clone** in clone/clone.js file

**Comment by erdii** [comment]
This is acutally pretty strange.... can  you shoot me a repo link with an example that I can execute and debug through?

**Comment by opusvega** [comment]
I found the issue. It was just because of deep nested object. I am trying to store following object as a value.
```
var obj = {
     socket : socket,
     users : 0
}
```
where socket value is from socket.io
Even i tried JSON.strigify(obj) which gives circular reference error neither does obj.toString() works so as I could parse it but which again throws error. How could I store above deep object as value in my node cache?

**Comment by opusvega** [comment]
The main issue was in CircularJson structure of our socket object. It was resolved by using circular-json npm  module by which we stored the socket value as CircularJSON.stringify(obj). And as I told earlier that the set function was only working outside the io.on() well I found it was because I haven't tried storing circular object there. So it seems to me it was working there. Anyway thanks Erdii for your valuable time and help.
